apps/shell/tash_main.c

```diff
@@ -106,11 +106,24 @@ static void tash_remove_char(char *char_pos)
  */
 static char *tash_read_input_line(int fd)
 {
+	#define SKIP_NEXT_CHAR(n)    char_idx += (n)
+	#define PREV_CHAR            buffer[pos - 1]
+	#define CURR_CHAR            buffer[pos]
+	#define NEXT_CHAR            buffer[pos + 1]
+	#define NEXTNEXT_CHAR        buffer[pos + 2]
+
 	int bufsize = TASH_LINEBUFLEN;
 	int pos = 0;
-	int nbytes = 0;
-	int char_idx = 0;
+	int nbytes;
+	int char_idx;
 	bool is_tab_pressed = false;
+#if TASH_MAX_STORE > 0
+	bool is_esc_pressed = false;
```

Comment:
Why don't you use bit operation for these? if I were you I would do that.
But you can ignore this comment because of different style.

Reply:
Because they are stack variables and they are not deliverable, I don't know bit operation is better or not.